### ISSUE TITLE
Determine roll-out plan for Discourse site

### ISSUE THREAD

**Issue by jwflory** [FOSSRIT/tasks#78]
# Summary

Come up with strategy and plan for implementing Discourse site and using it in FOSS@MAGIC community

# Background

The Discourse site is online at [fossrit.community](https://fossrit.community/). Now that it is basically set up, we need to figure out its role and use inside the FOSS@MAGIC community.

Ideally, this would become the new "hub" of virtual discussion and FOSS news around the RIT community. This would replace our `@fedorahosted.org` mailing list. We previously discussed using it for courses in the FOSS minor, like Free Culture with Prof. Ray.

# Details

To move forward, we need to do the following:

* Map out how we want to use Discourse
* Introduce it to other faculty teaching FOSS courses
* Figure out what to do with our mailing list

# Outcome

* Better web presence for FOSS @ RIT (scores better on search engines than mailing list)
* Encourage more participation by people with different preferences (e.g. Discourse's mailing list mode vs. using a web browser)

**Comment by ritjoe** [comment]
On Wed, Jan 23, 2019 at 02:59:18PM -0800, Justin W. Flory wrote:


> Ideally, this would become the new "hub" of virtual discussion and FOSS news
> around the RIT community. This would replace our @fedorahosted.org mailing
> list. We previously discussed using it for courses in the FOSS minor, like Free
> Culture with Prof. Ray.

I recall having seen, from various corners, claims that Discourse could replace a mailing list because it has some mail integrations.  This goes back to before my having used it at all.

With that in mind, I've created accounts in a couple of instances: The one the FSF put up recently, and the one associated with some of the Sustain conference efforts.

There may be settings for email preferences that I have yet to ferret out, but the default experience I see falls very far away from anything remotely resembling, or a replacement for, a mailing list. The emails I get from each instance show me a very shallow "teaser" view--I don't recally seeing anything resembling a threaded email discussion. It may be that threaded discussions just aren't happening, or it may be that the email interface is truncating what it shows me.

The nature of forums, in contrast to email lists, is that forums are a very "pull" kind of medium: One has to visit the web interface to participate (either repointing one's browser, perhaps from a bookmark, or keeping a persistent tab open and arranging for it to open again when restarting the browser).

The emails I've been getting look like an attempt to compensate for this in a way by drawing one back to the web interface. They do not look like an alternative way to interact with the system via email.

I'll contrast this behavior to Mailman 3/Hyperkitty, which offers a web interface but a first-class experience to someone interacting through a mail client. Similarly, my response to this github issue is being composed in my mail client of choice, and while GitHub's mail interface to issues and commits can be very noisy and doesn't obviate *entirely* the need to use the web interface occasionally, it does provide a pretty good experience.

I hope this captures the sort of take I had to share about this during in-person conversation last week. What remains is to see if there are other settings available that make the email experience something deeper than the "this has what you're missing on Facebook/Pinterest/Nextdoor" look it has for me now.

-- 
Joe

**Comment by jwflory** [comment]
* Partially blocked by meeting with Jenn about affiliate content and acceptable brand use (see #80)
* @jwflory to meet with Prof. Amit Ray today (Feb. 6th) about using Discourse in ENGL-450 course

**Comment by jwflory** [comment]
I met with Prof. Ray today and took a few implementation notes:

## Taxonomy

* New categories per post; option to use tags to sort content across semesters (e.g. `2185`, `2191`)
* Static content (i.e. references or cross-semester content)
  * Pinned, closed threads to distinguish static content
  * Appears at top of forum, distinguished from other posts as a pinned post
* Start simple: one category for the class, use tags for greater level of sorting
  * If requirements change later, it's easy to split out content into a new sub-category

## Implementation

* Prof. Ray has MediaWiki-formatted content across the last decade, relevant to this course
* @jwflory: Convert content from MediaWiki to markdown using `pandoc` to help populate the new category
* Create a new `ENGL-450` sub-category to use as a sandbox before the class runs next fall

**Comment by jwflory** [comment]
The [ENGL-450 category](https://fossrit.community/c/rit/engl-450) now exists. :tada: I followed up with Prof. Ray about migrating the existing MediaWiki content into Discourse.

**Comment by jwflory** [comment]
I think Discourse is sufficiently "rolled out" now. We can determine any future policy or strategies in a new ticket as necessary. Closing as complete. :clapper:
